### PR TITLE
unique_ptr for HierarchyIntegrator

### PR DESCRIPTION
**Pull request by marshallrdavey**
see #579

### CODE REVIEW

File: src/IB/IBHierarchyIntegrator.cpp [hunk 1]
```diff
@@ -317,7 +318,8 @@ IBHierarchyIntegrator::initializeHierarchyIntegrator(Pointer<PatchHierarchy<NDIM
         d_p_ghostfill_alg = new RefineAlgorithm<NDIM>();
         d_p_ghostfill_op = nullptr;
         d_p_ghostfill_alg->registerRefine(d_p_idx, d_p_idx, d_p_idx, d_p_ghostfill_op);
-        registerGhostfillRefineAlgorithm(d_object_name + "::p", d_p_ghostfill_alg, d_p_phys_bdry_op);
+        std::unique_ptr<RefinePatchStrategy<NDIM> > d_p_phys_bdry_op_unique(d_p_phys_bdry_op);
```
Comment (drwells) [inline src/IB/IBHierarchyIntegrator.cpp]: We shouldn't prefix these with `d_` - that is reserved for member variables.

**Comment by marshallrdavey** [comment]
@drwells not sure about the documentation you wanted added, but the changes compile and run

**Comment by drwells** [comment]
We should document `d_u_phys_bdry_op` and `d_p_phys_bdry_op` to state that the base class is responsible for the objects since they are placed in `d_ghostfill_strategies`.

**Comment by marshallrdavey** [comment]
@drwells should this documentation be added to IBHierarchyIntegrator.h above where the variables are defined? I don't know how to edit changelog.

**Comment by drwells** [comment]
To edit the changelog just add a new file under `doc/news/changes/incompatibilities/` - you can look at some entries in `doc/news/changes/major` to get an idea of what to write.

**Comment by marshallrdavey** [comment]
@drwells I'll collate the commits once everything is good to go

**Comment by drwells** [comment]
This does break the interface slightly, but since that interface is not used in any of the examples I don't think its a big deal.

We really do need to change this interface since passing a non-unique pointer implies that we do not assume ownership, but we do - we are inviting double-delete errors with the current implementation.

**Pull request merged**